client/src/dojo/game/models/TutorialGrid.ts

```diff
@@ -0,0 +1,107 @@
+export interface Block {
+    value: number;
+    color: number;
+  }
+  
+export class TutorialGrid {
+    private grid: Block[][];
+    private width: number;
+    private height: number;
+  
+    constructor(width: number = 4, height: number = 4) {
+      this.width = width;
+      this.height = height;
+      this.grid = this.createEmptyGrid();
+    }
+  
+    private createEmptyGrid(): Block[][] {
+      return Array(this.height).fill(null).map(() => 
+        Array(this.width).fill(null).map(() => ({ value: 0, color: 0 }))
+      );
+    }
+  
+    public setBlock(row: number, col: number, value: number, color: number): void {
+      if (this.isValidPosition(row, col)) {
+        this.grid[row][col] = { value, color };
+      }
+    }
+  
+    public getBlock(row: number, col: number): Block | null {
+      if (this.isValidPosition(row, col)) {
+        return this.grid[row][col];
+      }
+      return null;
+    }
+  
+    private isValidPosition(row: number, col: number): boolean {
+      return row >= 0 && row < this.height && col >= 0 && col < this.width;
+    }
+  
+    public slideCube(fromRow: number, fromCol: number, toRow: number, toCol: number): boolean {
+      if (!this.isValidPosition(fromRow, fromCol) || !this.isValidPosition(toRow, toCol)) {
+        return false;
+      }
+  
+      const block = this.grid[fromRow][fromCol];
+      if (block.value === 0) {
+        return false; // Can't slide an empty space
+      }
+  
+      // Check if the move is valid (only horizontal or vertical)
+      if (fromRow !== toRow && fromCol !== toCol) {
+        return false;
+      }
+  
+      // Slide the cube
+      this.grid[toRow][toCol] = block;
+      this.grid[fromRow][fromCol] = { value: 0, color: 0 };
+  
+      return true;
+    }
+  
+    public applyGravity(): void {
```

Comment:
why implement it again? can't you use the implementation iwhere it's implemented in the first place?